save-backend/src/main/kotlin/com/saveourtool/save/backend/service/LnkContestExecutionService.kt

```diff
@@ -49,4 +54,20 @@ class LnkContestExecutionService(
      * with any [Contest]
      */
     fun findContestByExecution(execution: Execution) = lnkContestExecutionRepository.findByExecution(execution)?.contest
+
+    /**
+     * @param execution
+     * @param contestName
+     * @return [Mono] containing a created [LnkContestExecution] or `Mono.error` with code 404
+     */
+    fun createLink(execution: Execution, contestName: String) = blockingToMono {
+        contestRepository.findByName(contestName)
+    }
+        .filter { it.isPresent }
+        .map {
+            lnkContestExecutionRepository.save(
+                LnkContestExecution(execution = execution, contest = it.get())
+            )
+        }
+        .switchIfEmptyToNotFound()
```

Comment:
Are you sure it should be here but not before `map`?

Reply:
Doesn't really matter in this case, but I've changed for better readability